### PR TITLE
More multitenancy work

### PR DESCRIPTION
**Pull request by ghost**
Changes on top of #22.

### CODE REVIEW

File: decisiontree/tests/test_app.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+from .cases import DecisionTreeTestCase
+
+
+class BasicSurveyTest(DecisionTreeTestCase):
+
+    def setUp(self):
+        super(BasicSurveyTest, self).setUp()
+        self.survey = mommy.make('decisiontree.Tree', trigger='food')
+        link_models.TreeLink.all_tenants.create(
```
Comment (vkurup) [inline decisiontree/tests/test_app.py]: These could use the default manager. `link_models.TreeLink.objects.by_tenant(self.tenant).create(linked=self.survey)`. It doesn't matter too much, but it does enforce that a tenant should be provided, where the all_tenants manager doesn't.

File: decisiontree/utils.py [hunk 2]
```diff
@@ -1,5 +1,17 @@
...
+
+
+def get_survey(trigger, connection):
+    """Returns a survey only if it matches the connection's tenant."""
+    from decisiontree.multitenancy.utils import multitenancy_enabled
+    queryset = Tree.objects.filter(trigger__iexact=trigger)
+    if multitenancy_enabled():
+        tenant = connection.backend.tenantlink.tenant
+        queryset = queryset.filter(tenantlink__tenant=tenant)
+    return queryset[0] if queryset else None
```
Comment (vkurup) [inline decisiontree/utils.py]: I recently learned about the `first()` queryset method that does the same thing:
https://docs.djangoproject.com/en/1.7/ref/models/querysets/#first

**Comment by vkurup** [comment]
A bunch of nitpicky comments, but this looks great. :cactus:

**Comment by ghost** [comment]
Updated all. And fixed that ghostly test error. Thanks for the many helpful comments!

**Pull request merged**